City-of-Helsinki/berth-reservations
# VEN-552 | Boat places is_active

## Description :sparkles:
- Add a `is_active` field to `AbstractBoatPlace`
- Expose `is_active` through GQL for queries and mutations
- Add `is_active=True` to the list of criteria to count `number_of_free_places`
- Add an additional `pre-commit` hook to check for `print` statements

## Issues :bug:
### Closes :no_good_woman:
**[VEN-552](https://helsinkisolutionoffice.atlassian.net/browse/VEN-552):** Add "is_active" boolean field to AbstractBoatPlace model

## Testing :alembic:
### Automated tests :gear:️
```shell
$ pytest  resources/tests/test_resources_mutations.py::test_create_berth
$ pytest  resources/tests/test_resources_mutations.py::test_update_berth

$ pytest  resources/tests/test_resources_queries.py::test_get_berths
$ pytest  resources/tests/test_resources_queries.py::test_get_harbor_available_active_berths
```

### Manual testing :construction_worker_man:
1. Query the `number of free places` on a Harbor
```graphql
query HARBOR {
  harbor(id: "SGFyYm9yTm9kZTowMWI3NWVmNy04MmQ1LTRiMzgtOGFhMS1iN2JkZGIyYjIwMTk=") {
    id
    properties {
      name
      numberOfPlaces
      numberOfFreePlaces
    }
  }
}
```
2. Mark a berth belonging to that harbor as `is_active=False` either throught a GQL mutation or through the admin
3. Check again the `number of free places`, it should have one free place less

## Screenshots :camera_flash:
**Berth admin**
<img width="526" alt="image" src="https://user-images.githubusercontent.com/15201480/79318716-02b76880-7f10-11ea-9f3b-201c11b993e6.png">

**BerthNode docs**
<img width="352" alt="image" src="https://user-images.githubusercontent.com/15201480/79318753-0ea32a80-7f10-11ea-9e8d-8285afc4b8bd.png">
